packages/website/src/components/lib/jsonSchema.ts

```diff
@@ -3,33 +3,106 @@ import type * as ts from 'typescript';
 
 import type { CreateLinter } from '../linter/createLinter';
 
+const defaultRuleSchema: JSONSchema4 = {
+  type: ['string', 'number'],
+  enum: ['off', 'warn', 'error', 0, 1, 2],
+};
+
+/**
+ * Add the error level to the rule schema items
+ *
+ * if you encounter issues with rule schema validation you can check the schema by using the following code in the console:
+ * monaco.languages.json.jsonDefaults.diagnosticsOptions.schemas.find(item => item.uri.includes('typescript-eslint/consistent-type-imports'))
+ * monaco.languages.json.jsonDefaults.diagnosticsOptions.schemas.find(item => item.uri.includes('no-unused-labels'))
+ * monaco.languages.json.jsonDefaults.diagnosticsOptions.schemas.filter(item => item.schema.type === 'array')
+ */
+export function getRuleJsonSchemaWithErrorLevel(
+  name: string,
+  ruleSchema: JSONSchema4 | JSONSchema4[],
+): JSONSchema4 {
+  if (Array.isArray(ruleSchema)) {
+    return {
+      type: 'array',
+      items: [defaultRuleSchema, ...ruleSchema],
+      minItems: 1,
+      additionalItems: false,
+    };
+  }
+  // TODO: delete this once we update schemas
+  // example: ban-ts-comment
+  if (Array.isArray(ruleSchema.prefixItems)) {
+    const { prefixItems, ...rest } = ruleSchema;
+    return {
+      ...rest,
+      items: [defaultRuleSchema, ...(prefixItems as JSONSchema4[])],
+      maxItems: ruleSchema.maxItems ? ruleSchema.maxItems + 1 : undefined,
+      minItems: ruleSchema.minItems ? ruleSchema.minItems + 1 : 1,
+      additionalItems: false,
+    };
+  }
+  // example: explicit-member-accessibility
+  if (Array.isArray(ruleSchema.items)) {
+    return {
+      ...ruleSchema,
+      items: [defaultRuleSchema, ...ruleSchema.items],
+      maxItems: ruleSchema.maxItems ? ruleSchema.maxItems + 1 : undefined,
+      minItems: ruleSchema.minItems ? ruleSchema.minItems + 1 : 1,
+      additionalItems: false,
+    };
+  }
+  if (typeof ruleSchema.items === 'object' && ruleSchema.items) {
+    // example: naming-convention rule
+    return {
+      ...ruleSchema,
+      items: [defaultRuleSchema],
+      additionalItems: ruleSchema.items,
+    };
+  }
+  // example eqeqeq
+  if (Array.isArray(ruleSchema.anyOf)) {
+    return {
+      ...ruleSchema,
+      anyOf: ruleSchema.anyOf.map(item =>
+        getRuleJsonSchemaWithErrorLevel(name, item),
+      ),
+    };
+  }
+  // example logical-assignment-operators
+  if (Array.isArray(ruleSchema.oneOf)) {
+    return {
+      ...ruleSchema,
+      oneOf: ruleSchema.oneOf.map(item =>
+        getRuleJsonSchemaWithErrorLevel(name, item),
+      ),
+    };
+  }
+  if (typeof ruleSchema !== 'object' || Object.keys(ruleSchema).length) {
+    console.error('unsupported rule schema', name, ruleSchema);
+  }
```

Comment:
this error is going to be printed only if schema is invalid eg, 
currently there is only one rule that is reported here `@typescript-eslint/lines-between-class-members` and its correctly marked as invalid schema

![image](https://user-images.githubusercontent.com/625469/232215623-6e3d94d8-a1c1-4740-9e1a-bc7c4eaa90f6.png)